airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/load/dataflow/pipeline/DataFlowPipeline.kt

```diff
@@ -6,31 +6,25 @@ package io.airbyte.cdk.load.dataflow.pipeline
 
 import io.airbyte.cdk.load.dataflow.config.MemoryAndParallelismConfig
 import io.airbyte.cdk.load.dataflow.stages.AggregateStage
-import jakarta.inject.Named
-import jakarta.inject.Singleton
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.buffer
 import kotlinx.coroutines.flow.flowOn
 import kotlinx.coroutines.flow.map
 import kotlinx.coroutines.flow.onCompletion
-import kotlinx.coroutines.flow.onStart
 import kotlinx.coroutines.flow.transform
 
-@Singleton
 class DataFlowPipeline(
     private val input: Flow<DataFlowStageIO>,
-    @Named("parse") private val parse: DataFlowStage,
-    @Named("aggregate") private val aggregate: AggregateStage,
-    @Named("flush") private val flush: DataFlowStage,
-    @Named("state") private val state: DataFlowStage,
-    private val startHandler: PipelineStartHandler,
+    private val parse: DataFlowStage,
```

Comment:
You have three `DataFlowStage`, is the type expected to be the same? Asking because I see an `AggregateStage`